lib/processors/versionInfoGenerator.js

```diff
@@ -14,6 +16,387 @@ function getTimestamp() {
+	if (typeof embeddedBy === "undefined") {
+		log.verbose("  component doesn't declare 'sap.app/embeddedBy', don't list it as 'embedded'");
+		return false;
+	}
+	if (typeof embeddedBy !== "string") {
+		log.error(
+			"  component '%s': property 'sap.app/embeddedBy' is of type '%s' (expected 'string'), " +
+			"it won't be listed as 'embedded'", componentPath, typeof embeddedBy
+		);
+		return false;
+	}
+	if ( !embeddedBy.length ) {
+		log.error(
+			"  component '%s': property 'sap.app/embeddedBy' has an empty string value (which is invalid), " +
+			"it won't be listed as 'embedded'", componentPath
+		);
+		return false;
+	}
+	let resolvedEmbeddedBy = posixPath.resolve(componentPath, embeddedBy);
+	if ( resolvedEmbeddedBy && !resolvedEmbeddedBy.endsWith("/") ) {
+		resolvedEmbeddedBy = resolvedEmbeddedBy + "/";
+	}
+	if ( libraryPathPrefix === resolvedEmbeddedBy ) {
+		log.verbose("  component's 'sap.app/embeddedBy' property points to library, list it as 'embedded'");
+		return true;
+	} else {
+		log.verbose(
+			"  component's 'sap.app/embeddedBy' points to '%s', don't list it as 'embedded'", resolvedEmbeddedBy
+		);
+		return false;
+	}
+};
+
+/**
+ * Retrieves the manifest path of a subcomponent
+ *
+ * @param {string} filePath path to the manifest, e.g. "lib/x/manifest.json"
+ * @param {string} subPath relative sub path, e.g. "sub"
+ * @returns {string} manifest path, e.g. "lib/x/sub/manifest.json"
+ */
+const getManifestPath = (filePath, subPath) => {
+	const folderPathOfManifest = filePath.substr(0, filePath.length - "manifest.json".length) + subPath;
+	return posixPath.resolve(folderPathOfManifest + "/manifest.json");
+};
+
+/**
+ * Represents dependency information for a library.
+ * Dependencies can be resolved recursively using <code>#resolve</code>
+ * and are stored then in <code>libsResolved</code>
+ */
+class DependencyInfo {
+	/**
+	 *
+	 * @param {ManifestLibraries} libs
+	 * @param {string} name library name, e.g. "lib.x"
+	 */
+	constructor(libs, name) {
+		this.libs = libs;
+		this.name = name;
+
+		/**
+		 * contains as key the library name and as value an object with an optional lazy property
+		 *
+		 * @type {ManifestLibraries}
+		 */
+		this.libsResolved = Object.create(null);
+		this.wasResolved = false;
+	}
+
+	/**
+	 * Add library to libsResolved and if already present
+	 * merge lazy property
+	 *
+	 * @param {string} libName library name, e.g. "lib.x"
+	 * @param {boolean} lazy
+	 * @returns {{lazy: boolean}} the added library
+	 */
+	addResolvedLibDependency(libName, lazy) {
+		let alreadyResolved = this.libsResolved[libName];
+		if (!alreadyResolved) {
+			alreadyResolved = Object.create(null);
+			if (lazy) {
+				alreadyResolved.lazy = true;
+			}
+			this.libsResolved[libName] = alreadyResolved;
+		} else {
+			// siblings if sibling is eager only if one other sibling eager
+			alreadyResolved.lazy = alreadyResolved.lazy && lazy;
```

Comment:
this is already handled by:
https://github.com/SAP/ui5-builder/pull/554/files/b9da72db5dec100900c85b27ce69c2cee73a1a88#diff-5fc9387f7e46413dbf8c8988ee00af7a121520ae98e152f8c107d782a77c1ee6R65
it can only be `true`

Reply:
Not only by that line, but with the help of several others and by the nature of the && operator and by JSON.stringify which doesn't stringify an undefined value...

**But you're right**. I indeed wondered why the tests didn't show any unexpected `lazy:false` values but was too lazy to investigate this.

Reply:
I would leave it untouched then.